url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/StandAloneSimulation.java

```diff
@@ -0,0 +1,527 @@
+  		for(Properties prop : sfp.propActivities) 
+  		{
+			String activityName = prop.getProperty("name");
+			String activityID = prop.getProperty("id");
+			String activityDesc = prop.getProperty("description") != null ? prop.getProperty("description") : "";
+			String activityType = prop.getProperty("type") != null ? prop.getProperty("type") : "";
+			Activity act = new Activity.Builder(activityID, activityName, activityDesc, activityType, simParams).build();
+			activities.put(activityID, act);
+			Person person = people.get(prop.getProperty("person").trim());
+			person.addActivity(act);
+  		}
+  		
+  		// Create the activity models 
+  		for(Properties prop : sfp.propActModels) 
+  		{
+  			String id = prop.getProperty("id");
+  			String actmodDayType = prop.getProperty("day_type");
+  			boolean shiftable = Boolean.parseBoolean((prop.getProperty("shiftable") != null ? prop.getProperty("shiftable").trim() : "false")); 	
+  			boolean exclusive = Boolean.parseBoolean((prop.getProperty("exclusive") != null ? prop.getProperty("exclusive").trim() : "true")); 	 
+  			String temp = prop.getProperty("containsAppliances");
+  			String[] containsAppliances = temp.split(",");
+  			
+  			Activity act = activities.get(prop.getProperty("activity").trim());
+  			
+  			String distrType = prop.getProperty("start_distrType");
+  			ProbabilityDistribution actModelStart;
+			try {
+				actModelStart = constructDistribution(distrType, prop, "start");
+				act.addStartTime(actmodDayType, actModelStart);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+
+			distrType = prop.getProperty("duration_distrType").trim();
+  			ProbabilityDistribution actModelDuration;
+  			try {
+  				actModelDuration = constructDistribution(distrType, prop, "duration");
+  				act.addDuration(actmodDayType, actModelDuration);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+  			
+  			distrType = prop.getProperty("repetitions_distrType");
+  			ProbabilityDistribution actModelTimes;
+  			try {
+  				actModelTimes = constructDistribution(distrType, prop, "repetitions");
+  				act.addTimes(actmodDayType, actModelTimes);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+  			
+  			act.addShiftable(actmodDayType, shiftable);
+  			act.addConfig(actmodDayType, exclusive);
+  			
+  			act.addAppliances(containsAppliances, appliances, actmodDayType);
+  			
+//  			for(int m = 0; m < containsAppliances.length; m++) {
+//  				String containAppId = containsAppliances[m].trim();
+//  				Appliance app  = appliances.get(containAppId);
+//  				act.addAppliance(actmodDayType,app,1.0/containsAppliances.length);
+//  			}
+  		}
+		
+		this.simulationWorld = simParams;
+  		this.pricing = pricPolicy;
+  		this.baseline_pricing = pricPolicyB;
+  		
+  		
+  		return installations;
+	}
+	
+	
+	private PricingPolicy constructPricingPolicy(String pricingType, PricingPolicy.Builder builderPP) throws Exception
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.